provider_packages/SETUP_TEMPLATE.py.jinja2

```diff
@@ -0,0 +1,101 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# NOTE! THIS FILE IS AUTOMATICALLY GENERATED AND WILL BE\
+# OVERWRITTEN WHEN RUNNING
+#
+# ./breeze prepare-provider-readme
+#
+# IF YOU WANT TO MODIFY IT, YOU SHOULD MODIFY THE TEMPLATE
+# `SETUP_TEMPLATE.py.jinja2` IN the `provider_packages` DIRECTORY
+
+"""Setup.py for the {{ PACKAGE_PIP_NAME }} package."""
+
+import io
+import logging
+import os
+import sys
+
+from os.path import dirname
+from setuptools import find_packages, setup
+
+logger = logging.getLogger(__name__)
+
+version = '{{RELEASE_NO_LEADING_ZEROS}}'
+
+my_dir = dirname(__file__)
+
+try:
+    with io.open(os.path.join(my_dir,
+                              '{{ PROVIDER_PATH }}/{{ README_FILE }}'),
+                 encoding='utf-8') as f:
+        long_description = f.read()
+except FileNotFoundError:
+    long_description = ''
+
+
+def do_setup(version_suffix=''):
+    """
+    Perform the package setup.
+    """
+    setup(
+        name='{{ PACKAGE_PIP_NAME }}',
+        description='{{ PROVIDER_TYPE }} package {{ PACKAGE_PIP_NAME }} for Apache Airflow',
+        long_description=long_description,
+        long_description_content_type='text/markdown',
+        license='Apache License 2.0',
+        version=version + version_suffix,
+        packages=find_packages(
+            include=['airflow.providers.{{ PROVIDER_PACKAGE_ID }}*']),
+        zip_safe=False,
+        install_requires={{ INSTALL_REQUIREMENTS }},
+        setup_requires={{ SETUP_REQUIREMENTS }},
+        extras_require={{ EXTRAS_REQUIREMENTS }},
+        classifiers=[
+            'Development Status :: 5 - Production/Stable',
+            'Environment :: Console',
+            'Environment :: Web Environment',
+            'Intended Audience :: Developers',
+            'Intended Audience :: System Administrators',
+            'License :: OSI Approved :: Apache Software License',
+            'Programming Language :: Python :: 3.6',
+            'Programming Language :: Python :: 3.7',
+            'Programming Language :: Python :: 3.8',
+            'Topic :: System :: Monitoring',
+        ],
+        author='Apache Software Foundation',
+        author_email='dev@airflow.apache.org',
+        url='http://airflow.apache.org/',
+        download_url=(
+            'https://dist.apache.org/repos/dist/release/airflow/{{ PROVIDERS_FOLDER }}' + version),
+        python_requires='~=3.6',
+    )
+
+
+if __name__ == "__main__":
+    suffix=''
+    if len(sys.argv) == 1:
+        print("ERROR! Missing first param", file=sys.stderr)
+        sys.exit(1)
+    if sys.argv[1] == "--version-suffix":
+        if len(sys.argv) < 3:
+            print("ERROR! --version-suffix needs parameter!", file=sys.stderr)
+            sys.exit(1)
+        suffix = sys.argv[2]
+        sys.argv = [sys.argv[0]] + sys.argv[3:]
+    do_setup(version_suffix=suffix)
```

Comment:
This should not be in the generated setup.py -- it means you can't take the dist and run setup.py without again providing this suffix. It should instead be "baked" in, so that the setup.py in the dist is fully static and operates like a plain module.

Reply:
Sure I can bake it in for alphas, no problem. But I will leave an option to modify it manually for RCs. I use the same setup approach for backports and I prefer to add rc suffixes when generating the RCs rather than manually modifying the sources during the builds as it happens currently. Effectively we are releasing tio PyPI packages that haave slightly modified sources vs. the ones we actually put in SVN. I will rename the switch as "--pypi-rc-suffix" though to be perfectly clear about it's usage.

Reply:
-1 from me, sorry -- hte dists we release, be it pypi or for svn, should be plain old non-configurable dists.

When releasing apache-airflow, we produce two versions, one for SVN and voting, and one for pypi.

Reply:
And Just to clarify - for A1/B1 the version is already part of the sources, when readmes/setup files are generated. This parameter from the beginning was only supposed to be used for RC candidates that need different RC in sources and different in PyPI. I just simply clarify it by changing its name and adding extra comments.

Reply:
Looks like, while I was INTENDING to do this, I did not. It was a good call @ashb :).